src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -235,12 +321,91 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
     } else if (key == CONFIG_KEY_INTERNAL(CPU_THREADS_PER_STREAM)) {
         return {std::to_string(_threadsPerStream)};
+    } else if (key == CONFIG_KEY_INTERNAL(BIG_CORE_STREAMS)) {
+        return {std::to_string(_big_core_streams)};
+    } else if (key == CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)) {
+        return {std::to_string(_small_core_streams)};
+    } else if (key == CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)) {
+        return {std::to_string(_threads_per_stream_big)};
+    } else if (key == CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)) {
+        return {std::to_string(_threads_per_stream_small)};
+    } else if (key == CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)) {
+        return {std::to_string(_small_core_offset)};
     } else {
         IE_THROW() << "Wrong value for property key " << key;
     }
     return {};
 }
 
+int IStreamsExecutor::Config::LimitHybridStreams(const int num_streams, const int num_threads) {
+    const auto core_types = custom::info::core_types();
+    const auto num_big_cores_phys = getNumberOfCPUCores(true);
+    const auto num_small_cores =
+        custom::info::default_concurrency(custom::task_arena::constraints{}.set_core_type(core_types.front()));
+    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+
+    // limit streams to reasonable range
+    int streams = num_streams > base_streams_total ? base_streams_total : std::max(1, num_streams);
+    // if num_threads is set, streams must not be greater than num_threads
+    if (num_threads > 0 && streams > num_threads) {
+        streams = num_threads;
+    }
+    if (streams != num_streams) {
+        OPENVINO_WARN << "Number of streams is updated from " << num_streams << " to " << streams;
+    }
+    return streams;
+}
+
+void IStreamsExecutor::Config::UpdateHybridCustomThreads(Config& config,
+                                                         const int num_big_cores_phys,
+                                                         const int num_small_cores) {
+    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+    const auto num_cores_phys = num_big_cores_phys + num_small_cores;
+
+    config._small_core_offset = num_big_cores_phys * 2;
+    // if nthreads is set, calculate hybrid aware parameters by nthreads
+    if (config._threads) {
+        // limit threads to reasonable range
+        auto threads = config._threads > num_cores_phys ? num_cores_phys : config._threads;
+        // all threads are placed on big cores
+        if (threads <= num_big_cores_phys) {
+            config._big_core_streams = config._streams;
+            config._threads_per_stream_big = std::max(1, threads / config._streams);
+        } else {
+            // big cores first, then small cores
+            // big_streams = all_streams * big_cores / base_streams
+            config._big_core_streams =
+                config._streams * num_big_cores_phys % base_streams_total == 0
```

Comment:
If update `base_streams_total` as my previous comments, here should be `num_big_cores_phys * 2 % base_streams_total ` and `num_big_cores_phys * 2 / base_streams_total ` in following 3 lines.